src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -432,6 +439,6 @@ def deploy(self) -> DiffResult:
             self._apply_package_scripts()
 
             # 3. Upload files from deploy root local folder to the above stage
-            diff = self.sync_deploy_root_with_stage(self.package_role)
+            diff = self.sync_deploy_root_with_stage(self.package_role, files_to_stage)
```

Comment:
nit: files_to_deploy? stage is a noun in snowflake, not a verb (here and elsewhere).

Reply:
Agreed with the name change, it might also make the "filtering" property clearer if it's a lambda that decides whether or not to allow this file to be synced rather than the list of files to be synced (which makes it less clear what happens when a file appears in this list but not in the diff).